modules/internal/String.chpl

```diff
@@ -427,6 +427,75 @@ module String {
       return ret;
     }
 
+    /* This function takes an array of stings as input.
+
+    returns:  The longest common prefix path of all the input paths
+    */
+    proc commonPrefix(paths: []): string {
+
+    var prefix: string = "";    // result string
+    var inputLength = paths.size;   // size of input array
+    var firstPath = paths[1];
+    var delimiter: string;
+    var cnt: int = 1;
+    var pos: int ;    // rightmost index of common prefix
+    var minPathLength: int ;
+
+    // if input is empty, return empty string.
+    // if input is just one string, return that string as longest common prefix path.
+
+    if inputLength == 0 then {
+      return prefix;
+    } else if inputLength == 1 then{
+      return firstPath;
+    }
+
+    // finding delimiter to split the paths.
+
+    if firstPath.find("\\", 1..firstPath.length) == 0 then {
+      delimiter = "/";
+    } else {
+      delimiter = "\\";
+    }
+    cnt = firstPath.count(delimiter);
+
+    var prefixArray:[1..cnt+1] string;    // array of resultant prefix string
+    prefixArray = firstPath.split(delimiter, -1, false);
+
+    pos = prefixArray.size;
+    minPathLength = prefixArray.size;
+
+    for i in 2..inputLength do {
+
+      cnt = paths[i].count(delimiter);
+      var tempArray:[1..cnt+1] string;    // temporary array storing the current path under consideration
+      tempArray = paths[i].split(delimiter, -1, false);
+
+      var minimum = min(prefixArray.size, cnt+1);
+      if minimum < minPathLength then {
+        minPathLength = minimum;
+      }
+
+      for (itr, ind) in zip(1..minimum, 1..minimum) do {
+        if tempArray[itr]!=prefixArray[ind] then {
+          pos = ind;
+          break;
+        }
+      }
+    }
+
+    if pos < prefixArray.size then {
+      prefixArray.remove(pos..prefixArray.size);
+      prefixArray.push_back(" ");
+    }   else {
+      prefixArray.remove(minPathLength+1..prefixArray.size);
```

Comment:
I'm not sure I understand the purpose of this `else` branch.  Why would you remove something if the first path is entirely a prefix of all the other ones?

Reply:
The else part is present because if the pos was never updated then it must be the case that the one of the path (which is not the first path in the array) could have been subset of the all the paths provided, thus we remove the array elements from minPathLength+1.

Reply:
Oh, I see.  Because you've had to limit the iteration of the `for` loop to only traverse the minimum of the two paths being compared, it is possible not to have updated `pos`.  Got it.  I think that would be worth a comment explaining, since I found it confusing to follow when just looking at the code